packages/fiori/src/ShellBar.ts

```diff
@@ -274,6 +279,23 @@ const HANDLE_RESIZE_DEBOUNCE_RATE = 200; // ms
 	},
 })
 
+/**
+ * Fired, when the search button is activated.
+ * <b>Note:</b> You can prevent expanding/collapsing of the search field by calling <code>event.preventDefault()</code>.
+ *
+ * @allowPreventDefault
+ * @param {HTMLElement} targetRef dom ref of the activated element
+ * @param {Boolean} isVisible whether the search field is visible
+ * @public
+ */
+
+@event("search-button-click", {
```

Comment:
Hi there, the event decorator became generic recently, so that we can set the event detail type and have type checking of the event detail - see the rest of the events.

For example:
```ts
@event<ShellBarProductSwitchClickEventDetail>("product-switch-click", {
```